docs: update guides + clean-up basic crawler api docs (#179)
apify/apify-ts: docs/guides/got_scraping.mdx

---
id: got-scraping
title: Got Scraping
---

## Intro

> If we're using Basic Crawler, we have to send requests manually. In order to do this, We need to call `sendRequest()` like this:

Reviewer: ```suggestion
When using `BasicCrawler`, we have to send the requests manually. In order to do this, we can use the context-aware `sendRequest()` function:
```

By default, it's the URL of current task. However you can override this with a `string` or a `URL` instance if necessary.

> *More details: [Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#url).*

Reviewer: maybe its better this way? idk

```suggestion
*More details in [Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#url).*
```

Author: I was thinking about this one, both sounds good to me, will update to `in` with next PR :)

> *More details: [Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#url).*

Reviewer 2: What's the rationale behind this change? I don't mind, just asking.

Author: Originally @B4nan mentioned this `you` -> `we` thing on Slack, so basically I tried to change `[you] see more details` to something less direct/addressed 🤷‍♂️

But in the end, we're not doing it...

Reviewer 2: 😂 Ok I get it now :D That's kinda dumb because there's often `Read more` in other projects' docs.

Like I suggested, `you` refers explicitly to the developer reading the docs, so if *they* are given an option then the steps depend on *their* path *they* have chosen. So `you` would be appropriate. https://docs.oracle.com/en/java/

> Java SE lets you develop

There's an implied choice. You can choose Java or any other language.

OTOH `we` refers to both the developer and the project, there may be a thing that applies to both persons. For example a tutorial. https://docs.oracle.com/search/?q=we&category=java&product=en%2Fjava

> We will run the HelloWorld sample

Author: I think that's exactly what we're doing in the end 😄 But this PR was already merged before :/

});
```

*More details:*
- *[Got documentation](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#cookiejar)*
- *[Tough Cookie documentation](https://github.com/salesforce/tough-cookie#cookiejarstore-options)*

### `retry.limit`

This option specifies the maximum number of `Got` retries.

> By default, `retry.limit` is set to `0`. This is because `Crawlee` has its own (complicated enough) retry management.

Reviewer: ```suggestion
By default, `retry.limit` is set to `0`. This is because Crawlee has its own (complicated enough) retry management.
```
